src/services/RetroGenerationService.ts

```diff
@@ -84,36 +87,36 @@ class RetroGenerationService {
                     defectsDetails =
                       defectsDetails +
                       " " +
-                      testType.defects[key].deficiencyRef +
+                      testType[0].defects[0].deficiencyRef +
```

Comment:
Won't these 4 always refer to the very first defect in the array, even if it was on defect 2+?

Opposed to previously where it was referring to the [key] in Object.keys(testType.defects)

Reply:
Yes, good spot - will change that now